### ISSUE TITLE
Move extensions to separate output

### ISSUE THREAD

**Issue by jtojnar** [NixOS/nixpkgs#32817]
Some packages build other plug-ins, modules or extensions for external software like Nautilus, GNOME Shell or even GTK (https://github.com/NixOS/nixpkgs/issues/18559#issuecomment-352232675). These plug-ins make sense only for a subset of users, for example, Dejà Dup plug-in for Nautilus is probably not needed in XFCE (they use Thunar instead of Nautilus).

The following new outputs could be introduced to derivations:

* `nautilusExtension`: `lib/nautilus/extensions-3.0` (e.g. `deja-dup`)
* `gnomeShellExtension`: `share/gnome-shell/extensions` (e.g. `gnome3.pomodoro`, `gnome3.gpaste`)
* `mozillaPlugin`: `lib/mozilla/plugins` (e.g. `gnome3.evince`)
* `gtk3Module`: `lib/gtk-3.0/modules` (e.g. `gnome3.caribou`)
* `gtk3ImModule`: `lib/gtk-3.0/immodules` (e.g. `ibus`)

There are also some paths like `share/applications/gnome-control-center/keybindings` or `share/gnome-shell/search-providers` but those only contain simple XML files so moving them out it probably more trouble than it’s worth.

**Comment by vcunat** [comment]
Mozilla has killed plugins, so that's one less.  In general the splits are probably worth only in cases with a significant difference in closure...

**Comment by FRidh** [comment]
I propose non-FHS outputs have a standard prefix so as not to clutter the resulting attribute set. Would it work to prefix them with `output`, considering we also use `outputBin` and such in the implementation of multiple outputs.

**Comment by jtojnar** [comment]
I do not understand. Is not the `output*` only used for redirecting one output to another?

**Comment by FRidh** [comment]
I don't know the implementation details, that's why I am asking. Quickly tested creating an output `outputFoo` and things work as expected.

**Comment by jtojnar** [comment]
Yes, it will create the output with that name but the `output` prefix will not do anything special. On the other hand it will be confusable with the `output*` variables indirections – users might think that `outputFoo` is a variable that redirects `foo` output to a different one.
